app/helpers/proxmox_vm_config_helper.rb

```diff
@@ -27,11 +27,11 @@
 module ProxmoxVmConfigHelper
   def object_to_config_hash(vm, type)
     vm_h = ActiveSupport::HashWithIndifferentAccess.new
-    main_a = ['vmid']
-    main = vm.attributes.select { |key, _value| main_a.include? key }
+    main_a = ['vmid', 'type']
+    main = vm.attributes.select { |key, _value| main_a.include? key.to_s }
     main_a += ['templated']
     config = vm.config.attributes.reject do |key, _value|
-      main_a.include?(key) || Fog::Proxmox::DiskHelper.disk?(key) || Fog::Proxmox::NicHelper.nic?(key)
+      main_a.include?(key.to_s) || Fog::Proxmox::DiskHelper.disk?(key) || Fog::Proxmox::NicHelper.nic?(key)
```

Comment:
Should we also use `key.to_s` for `DiskHelper.disk?` and `NicHelper.nic?`, too`

Reply:
No, to match with the keys in a list we need to convert them to string but for disl and nic, it can take key as symbol as well.